### PR TITLE
Refactor release flow and use `changeset` to manage versioning

### PR DESCRIPTION
**Pull request by TuvalSimha**
- [x] Add `changeset` @TuvalSimha 
- [x] Add `renovate-bot` to the repo @kamilkisiela 
- [x] Add `NPM_TOKEN` secret @kamilkisiela 
- [x] Set `publishConfig` if needed @TuvalSimha 
- [x] Remove old release scripts @TuvalSimha 
- [x] Test on PR (using `changeset` file) @TuvalSimha 
- [ ] Test stable release (from `master`)  @TuvalSimha 
- [x] Add to `CONTRIBUTING.md` explaining about `changeset`, how to use it, etc...

### CODE REVIEW

**Review by PowerKiKi — CHANGES_REQUESTED** [review]
CONTRIBUTING.md contains some instructions on how to release. Could you please update them to explain briefly how this new system work and maybe also link to the full documentation of the tool?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -95,15 +96,20 @@ You can check your setup by running `yarn test`, it should test all packages and
...
-VERSION=1.2.3
-./scripts/version.js $VERSION
-yarn run deploy
-git add .
-git commit -m "Release v$VERSION"
-git tag $VERSION -m $VERSION
-git push
-```
-3. Create a release on https://github.com/kamilkisiela/apollo-angular/releases/new
+We are using `Changeset` to manage and publish our releasing flow.
```
Comment (TuvalSimha) [inline CONTRIBUTING.md:99]: Over kill it?

Comment (dotansimha) [reply CONTRIBUTING.md:99]: I think it's fine, but you can also just link to `changeset` repo and people will find there instructions 😆

**Review by dotansimha — APPROVED** [review]
LGTM. @PowerKiKi can you please confirm that the alpha version published from this PR (`4.2.1-alpha-20230125113210-0fc1de1c`) works fine? if so, we can merge and move forward

**Review by PowerKiKi — APPROVED** [review]
The released `4.2.0` and pre-released `4.2.1-alpha-20230125113210-0fc1de1c` are virtually identical. The only exception being the version in `package.json`. So I am pretty confident we can merge this.

Thanks !

**Comment by dotansimha** [comment]
Thanks @PowerKiKi ! 
@TuvalSimha I think we are good to go 🚀

**Pull request merged**